src/lib/stores.ts

```diff
@@ -63,20 +96,17 @@ export const handleSnsChange = async (name?: string, principal?: string) => {
 	if (!fetchedCanisters) return;
 
 	sns.reset();
-
+	inputAmount.reset();
 	if (name && principal) {
 		sns.setName(name);
 		const p = Principal.fromText(principal);
 		sns.setPrincipal(principal);
-		const account = await fetchedCanisters.boomerang.get_staking_account(
-			Principal.fromText(principal)
-		);
-		const hex = encodeIcrcAccount({
+		const account = await fetchedCanisters.boomerang.get_staking_account(p);
```

Comment:
This call is slow, and the other calls are blocked. Make the three calls in parallel or first fetch the balance as it's faster calls.

Reply:
made them in parallel